core/runtime/runtime_api/impl/parachain_host.cpp

```diff
@@ -149,6 +149,18 @@ namespace kagome::runtime {
     return *ref;
   }
 
+  outcome::result<std::vector<std::optional<CommittedCandidateReceipt>>>
+  ParachainHostImpl::candidates_pending_availability(
+      const primitives::BlockHash &block, ParachainId id) {
+    OUTCOME_TRY(ref,
+                candidates_pending_availability_.call(
+                    *executor_,
+                    block,
+                    "ParachainHost_candidates_pending_availability",
+                    id));
+    return *ref;
```

Comment:
Is the copy eliminated here/how large can this vector be? Maybe construct an outcome::result and move the vector there explicitly to avoid copying, if it can be large?

Reply:
Added explicit moving

Reply:
return std::move is typically a worse alternative, I mean construct a variable with type exactly matching the result type and move in its constructor, this practically guarantees RVO.

Reply:
I think move constructor for return value is guaranteed by this move. outcome::result<std::vector<std::optional<CommittedCandidateReceipt>>> res = std::move(*ref) is literally what's gonna happen. Anyway the vector will be moved efficiently.